src/MoBi.Core/Services/BuildingBlockVersionUpdater.cs

```diff
@@ -28,6 +30,21 @@ public void UpdateBuildingBlockVersion(IBuildingBlock buildingBlock, uint newVer
          if (buildingBlock == null) return;
          buildingBlock.Version = newVersion;
          publishSimulationStatusChangedEvents(buildingBlock);
+         publishModuleStatusChangedEvents(buildingBlock);
+      }
+
+      private void publishModuleStatusChangedEvents(IBuildingBlock buildingBlock)
+      {
+         // Building blocks that are not contained by modules do not need to refresh when modified
+         if (!_projectRetriever.Current.Modules.SelectMany(x => x.BuildingBlocks).Contains(buildingBlock))
+            return;
+
+         refreshModule(_projectRetriever.Current.Modules.Single(x => x.BuildingBlocks.Contains(buildingBlock)));
```

Comment:
I don't get the single here.

Reply:
I changed that because the building block knows who the module is.